src/irc.lua

```diff
@@ -15,6 +15,21 @@ irc.conn = function (irc_config)
     irc.connection:send(('USER %s * 8 :%s\r\n'):format(irc_config.ident, irc_config.gecos))
 end
 
+irc.action = function (target, msg)
+    irc.connection:send(('PRIVMSG %s :\x01ACTION %s\x01\r\n'):format(target, msg))
+end
+
+irc.CTCP_query = function (client, query)
+    -- VERSION, PING, SOURCE, CLIENTINFO, USERINFO, TIME, DCC are common
+    -- clients can also add their own.
+    -- https://tools.ietf.org/id/draft-oakley-irc-ctcp-01.html#rfc.appendix.A.2
+    irc.connection:send(('PRIVMSG %s :\x01%s\x01\r\n'):format(target, query))
+end
+
+irc.invite = function (recipient, channel)
```

Comment:
this might be the perfect opportunity to standardize this API's names (e.g., ``target`` or ``recipient``). Personally, I think I'd prefer ``target`` because it's a reasonable term regardless of talking about channel or user.

Reply:
I would honestly use target for things that can be a channel or a nick, and channel/nick for things that only apply to channels or  nicks, or maybe "client" for things that only apply to other clients (which are mostly other irc users), I'm not sure. But I totally agree the variable names should get standardized to something and I would be happy to do that. :)

Reply:
Oh, and maybe username always refers to the bot's username...?

Reply:
love that:

* ``target`` (could be anything you're targeting)
* ``channel`` (explicitly a channel)
* ``handle`` (explicitly a user)
* ``botnick`` ? (explicitly the bot's handle?)

Reply:
Does handle refer to the entire client/recipient name or just the nick...? IRC clients come in the format: `nick!identity@host/mask` identity is set by [identd](https://tools.ietf.org/html/rfc1413) but mostly only used for banning people to help differentiate multiple people on the same host.

Maybe it could be client, clientnick, clienthost, or user, username/usernick, userhost or even user/client, nick, host, idk there are a lot of combinations that would work. I like botnick and bothost for the bot, it makes it clear what the var refers to.

IRC having usernames, nicknames, identities and real names makes it more confusing and even in the rfc they interchange terms for the same thing between client/recipient/nick/user.

There is some variability in the script with message, msg, text, etc. for the custom text in replies. I think "message" reads nicer but it's longer to type and would increase line length.